hack/validate-migration.sh

```diff
@@ -0,0 +1,332 @@
+# discovered, it will be checked later. # No argument is required. Function
+# inspects changed files from current branch directly.
+list_migration_files() {
+    git diff --name-status "$(git merge-base HEAD "$DEFAULT_BRANCH")" | \
+        while read -r status origin_path; do
+            if ! is_migration_file "$origin_path"; then
+                continue
+            fi
+            case "$status" in
+                A)  # file is added
+                    echo "$origin_path"
+                    ;;
+                D | M)
+                    echo "It is not allowed to delete or modify existing migration file: $origin_path" >&2
+                    exit 1
+                    ;;
+                *)
+                    echo "warning: unknown operation for status $status on file $origin_path" >&2
+                    ;;
+            esac
+        done
+    return $?
+}
+
+K8S_TEST_NS=validate-migration-test
+declare -r K8S_TEST_NS
+
+# Check whether modified pipelines with applied migration is broken or not.
+# This check requires a created cluster with tekton installed.
+# An easy way to set up a local cluster is running `kind create cluster'.
+check_apply_in_real_cluster() {
+    if [ -z "$IN_CLUSTER" ]; then
+        return 0
+    fi
+    if ! kubectl get crd pipelines.tekton.dev -n tekton-pipelines >/dev/null 2>&1; then
+        echo "error: cannot find CRD pipeline.tekton.dev from cluster. Please create a cluster and install tekton." >&2
+        exit 1
+    fi
+    if kubectl get namespaces ${K8S_TEST_NS} >/dev/null 2>&1; then
+        kubectl delete pipeline --all -n ${K8S_TEST_NS} >/dev/null
+    else
+        kubectl create namespace ${K8S_TEST_NS}
+    fi
+    local -r apply_logfile=$(mktemp --suffix="-${FUNCNAME[0]}")
+    local -r modified_pipeline_files=$(find "${WORK_DIR}/pipelines" -type f -name "*.modified")
+    while read -r pl_file; do
+        if ! kubectl apply -f "$pl_file" -n ${K8S_TEST_NS} 2>"$apply_logfile" >/dev/null; then
+            echo "${FUNCNAME[0]}: failed to apply pipeline to cluster: $pl_file" >&2
+            cat "$apply_logfile" >&2
+            rm "$apply_logfile"
+            exit 1
+        fi
+    done <<<"${modified_pipeline_files}"
+    rm "$apply_logfile"
+}
+
+main() {
+    if git status --porcelain | grep -qv "^??"; then
+        echo "There are uncommitted changes. Please commit them and run again." >&2
+        exit 1
+    fi
+
+    if ! is_on_topic_branch; then
+        echo "Script must run on a topic branch rather than the main branch." >&2
+        return 1
+    fi
+
+    local -a migrations_files
+    mapfile -t migrations_files < <(list_migration_files)
+
+    if [[ ${#migrations_files[@]} -eq 0 ]]; then
+        echo "No migration."
+        exit 0
+    fi
+
+    local -r n=${#migrations_files[@]}
+    if [[ $n -gt 1 ]]; then
+        echo "error: found $n migration files. Please ensure to include a single migration file per time." >&2
+        exit 1
```

Comment:
I think it will often be necessary to include multiple migration files in a single PR. Especially for the buildah tasks, where you have to update 4+ tasks at the same time (buildah, buildah-oci-ta, buildah-remote, buildah-remote-oci-ta, buildah-*gb)

Reply:
Fixed in the new commit. PTAL.